README.md

```diff
@@ -1,70 +1,137 @@
-# Cluster Version Operator (CVO)
-
-## Building and Publishing CVO
-
-```sh
-./hack/build-image.sh && REPO=<your personal repo (quay.io/ahbinavdahiya | docker.io/abhinavdahiya)> ./hack/push-image.sh
+# Cluster Version Operator
+
+The Cluster Version Operator (CVO) is one of the [Cluster Operators](https://github.com/openshift/enhancements/blob/master/dev-guide/operators.md#what-is-an-openshift-clusteroperator)
+that run in every OpenShift cluster. CVO consumes an artifact called a _"release
+payload image,"_ which represents a given version of OpenShift and contains the
+manifests needed for all Cluster Operators to function. CVO reconciles
+the resources in the cluster to match the manifests in the release payload
+image. This way, CVO also implements _cluster upgrades_; after being provided
+a release payload image for a newer OpenShift version, CVO reconciles all
+Cluster Operators to their updated versions, and in turn, the Cluster Operators
+similarly update their operands.
+
+## OpenShift Upgrades
+
+For information about upgrading OpenShift clusters, please see the respective
+documentation:
+
+- [OKD Documentation: Updating clusters](https://docs.okd.io/latest/updating/index.html)
+- [Red Hat OpenShift Container Platform: Updating clusters](https://docs.openshift.com/container-platform/latest/updating/index.html)
+
+## `ClusterVersion` Resource
+
+Like other Cluster Operators, the Cluster Version Operator is configured by
+a Config API resource in the cluster: a [`ClusterVersion`](https://docs.openshift.com/container-platform/latest/rest_api/config_apis/clusterversion-config-openshift-io-v1.html):
+
+```console
+$ oc explain clusterversion
+  KIND:     ClusterVersion
+  VERSION:  config.openshift.io/v1
+
+  DESCRIPTION:
+       ClusterVersion is the configuration for the ClusterVersionOperator. This is
+       where parameters related to automatic updates can be set. Compatibility
+       level 1: Stable within a major release for a minimum of 12 months or 3
+       minor releases (whichever is longer).
+
+  FIELDS:
+     ...
+     spec	<Object> -required-
+       spec is the desired state of the cluster version - the operator will work
+       to ensure that the desired version is applied to the cluster.
+
+     status	<Object>
+       status contains information about the available updates and any in-progress
+       updates.
 ```
 
-1. This builds image locally and then pushes `${VERSION}` and `latest` tags to `${REPO}/origin-cluster-version-operator`.
-
-2. `${VERSION}` encodes the Git commit used to build the images.
-
-## Building release image using local CVO
+`ClusterVersion` resource follows the established [Kubernetes pattern](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#object-spec-and-status) where
+the `spec` sub-resource describes the desired state that CVO should achieve and
```

Comment:
nit: [only `status` is a strict subresource][1].  Can we use "property" or something instead, to avoid having an opinion on subresource-ness?

[1]: https://github.com/openshift/api/blob/94ac871f4a586d03b0ae11344f3604016065a458/config/v1/0000_00_cluster-version-operator_01_clusterversion.crd.yaml#L419-L420

Reply:
addressed in bf6cc77a13f199601f2f472ce21d6aa2af8d9c1c